lib/bap/bap_calls_reconstructor.ml

```diff
@@ -0,0 +1,87 @@
+open Core_kernel
```

Comment:
can you elaborate a little bit on the purposes and tasks of this module, and the overall algorithm description? It would be even nice to have this as a comment in the module.

Reply:
Ok, let's describe, what we did here. First of all, we gently touched the disassembler to allow fall-through edges leading to roots. And this affects the cfg only and doesn't affect subroutines on the IR level since our `sema lifter` doen't take into account whole program cfg at all, it deals only with subroutines cfgs, that's not sufficient for this task.

So we need to finish lifting procedure by taking into account every fall-through branch coming into subroutine entry. And by finish, I mean to edit a call site.

So algotithm, in general:

1) find the fall-through branch to a subroutine, i.e. to it's entry block
2) find a source in cfg, i.e. IR block where is came from
3) create and insert a call to the target subroutine after such block
4) if the source block contains no-returns then redirect them to 
   a brand new call, since the only reason it was created as noreturn
   is a lack of information about fall-through branches in `sema lifter`.

Will provide an example, like what is was before and how it looks like now.
E.g. `main` subroutine in `/bin/ls`. As far as I can see the only purpose is to push `R15`, that makes the name of this function a little bit too serious:
```
000003c6: sub main(main_argc, main_argv, main_result)
00015327: main_argc :: in u32 = RDI
00015328: main_argv :: in out u64 = RSI
00015329: main_result :: out u32 = RAX
000003c2: 
000003c3: v17136 := R15
000003c4: RSP := RSP - 8
000003c5: mem := mem with [RSP, el]:u64 <- v17136

```
after applying the algorithm in question we will see a continuation:
```
00015255: 
00015256: call @sub_402a02 with noreturn
```

Considering the place where to apply this transformation, we can still
to find a better place:
 - a separate pass
 - module in `bap_sema` library

Does it make any sence for you?